src/remote/index.js

```diff
@@ -99,9 +102,33 @@ export default class Remote {
       .catch(callback)
   }
 
-  // FIXME: Temporary stub so we can do some acceptance testing on file upload
+  async overwriteFileAsync (doc: Metadata, old: Metadata): Promise<RemoteDoc> {
+    const stream = await this.other.createReadStreamAsync(doc)
+    const updated = await this.remoteCozy.updateFileById(doc.remote._id, stream, {
+      contentType: doc.mime,
+      checksum: doc.checksum,
+      lastModifiedDate: new Date(doc.lastModification)
+    })
+
+    doc.remote._rev = updated._rev
+
+    return jsonApiToRemoteDoc(updated)
+  }
+
+  overwriteFile (doc: Metadata, old: Metadata, callback: Callback) {
+    this.overwriteFileAsync(doc, old)
+      .then(updated => callback(null, updated))
+      .catch(callback)
```

Comment:
> We should probably add a try/catch here

Isn't this already handled by the catch call on line 121?

> the await will throw an error, right?

Well, reading [this post](https://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html) I believed it wont, but according to both my unit and my manual tests, errors seems to correctly bubble up.

> The mix of callbacks/promises/async-await is not ideal.

Agreed. Then what about this?

```javascript
async overwriteFile (...) {
  try {
    const updated = await this.overwriteFileAsync(doc, old)
    callback(null, updated)
  } catch (err) {
    callback(err)
  }
}
```